src/charm.py

```diff
@@ -154,13 +177,87 @@ def _on_ingress_revoked(self, event: IngressPerAppRevokedEvent) -> None:
         if self.unit.is_leader():
             logger.info("This app no longer has ingress")
 
+    def _on_openfga_store_created(self, event: OpenFGAStoreCreateEvent) -> None:
+        """Handle openfga store created event."""
+        if not self._peers:
+            self.unit.status = WaitingStatus("Waiting for peer relation")
+            event.defer()
+            return
+
+        if self._get_openfga_store_info():
+            model_id = self._create_openfga_model()
```

Comment:
Doesn't this require the workload container/pebble to be ready? Should we check for it before calling this function?